### DISCUSSION TITLE
Newline in output deps map?

### DISCUSSION THREAD

**Discussion by zhming0** [babashka/neil#208] [Q&A]
Hi, I've been using neil add dep XYZ to add dependencies to my deps.edn file under `:deps`. I noticed that each new dependency gets added right after the last one on the same line. I was expecting each new addition to start on a new line for better readability. Is this the expected behavior, or is there a way to configure neil to append dependencies on new lines? 

I vaguely remember in the past it was working per my expectation, but my memory could be wrong.

**Comment by borkdude** [comment]
I ran into this myself too. This might be the case if there is no `:deps` entry yet and some edge case with https://github.com/borkdude/rewrite-edn

**Reply by borkdude** [reply]
If you can reproduce the situation with rewrite-edn, I'd me more than happy to receive an issue over there and I'll have a look.

**Reply by zhming0** [reply]
I managed to find a few test case that will break `rewrite-edn`, but it's not because of line break, instead it's because of newline's indentation. Don't know if they are related? 🤔 

```clojure
  (testing "zhming0 test"
    (is (= "
{:paths [\"src\"]
 :deps {org.clojure/clojure {:mvn/version \"1.11.1\"}
        com.s-exp/hirundo {:mvn/version \"0.1.34\"}}}"
           (str (r/assoc
                 (r/parse-string "
{:paths [\"src\"]
 :deps {org.clojure/clojure {:mvn/version \"1.11.1\"}}}")
                 'com.s-exp/hirundo {:mvn/version "0.1.34"})))))

```

From neil's perspective, this issue is pretty easy to reproduce though, any new `clj-new` project can verify this.

**Reply by borkdude** [reply]
OK, please give me a reproducible case, step by step, and I'll have a look.

**Reply by zhming0** [reply]
Sure, here is a (hopefully) reproducible steps: 
1. Run `docker run --rm -it --platform linux/amd64 homebrew/brew:4.2.15 /bin/bash`
2. Install neil and java: `brew install babashka/brew/neil`, `brew install java`.
3. `echo {:deps {}} > deps.edn`
4. `neil add dep com.s-exp/hirundo`
5. `neil add dep metosin/malli`
6. `cat deps.edn`

In my m1 max mac, step 3 ~ 6 in a `tmp/xx` folder will suffice to reproduce it. 
I added step 1 and 2 for better reproducibility.

**Comment by zhming0** [comment]
Thank you for fixing it! Looking forward to the new release ❤️ .

For other readers' context: fixed by https://github.com/babashka/neil/pull/210.
